components/js_lang/js_lang_core.cc

```diff
@@ -328,12 +340,95 @@ Js_isolate *Js_isolate::create() {
   return result;
 }
 
+void Js_isolate::Memory_manager::update_mem_stats(v8::HeapStatistics &stats) {
+  // Let us aggregate current memory usage counters for the isolate into
+  // global memory usage counters. To get actual global values, deduct values
+  // which were added to global counters on previous call of this method for
+  // the isolate.
+  s_total_heap_size += (stats.total_heap_size() - m_old_stats.total_heap_size);
+  s_used_heap_size += (stats.used_heap_size() - m_old_stats.used_heap_size);
+  // Using value from v8::HeapStatistics rather than directly from
+  // Memory_manager sounds more future-proof. For example, in future we might
+  // add size of SQL result objects to the amount of externally allocated
+  // memory (by reporting it to V8 using appropriate API).
+  s_external_memory += (stats.external_memory() - m_old_stats.external_memory);
+
+  // Save added counter values so they can be deducted on future calls or
+  // isolate destruction.
+  m_old_stats.total_heap_size = stats.total_heap_size();
+  m_old_stats.used_heap_size = stats.used_heap_size();
+  m_old_stats.external_memory = stats.external_memory();
+}
+
+void Js_isolate::Memory_manager::get_and_update_mem_stats() {
+  v8::HeapStatistics stats;
+  m_js_isolate->m_isolate->GetHeapStatistics(&stats);
+  update_mem_stats(stats);
+}
+
+std::atomic<int64_t> Js_isolate::Memory_manager::s_total_heap_size = 0;
+std::atomic<int64_t> Js_isolate::Memory_manager::s_used_heap_size = 0;
+std::atomic<int64_t> Js_isolate::Memory_manager::s_external_memory = 0;
+
+int Js_isolate::Memory_manager::show_total_heap_size(MYSQL_THD, SHOW_VAR *var,
+                                                     char *buff) {
+  var->type = SHOW_LONG;
```

Comment:
May be SHOW_LONGLONG here as we are using 64-bit values for counters.
Also, as far as I understood `SHOW_LONGLONG` is already an unsigned type as there is `SHOW_SIGNED_LONGLONG`. Please, correct me if I am wrong here.

Reply:
Indeed.